astropy/wcs/wcs.py

```diff
@@ -2446,6 +2448,26 @@ def sip_foc2pix(self, *args):
         """.format(docstrings.TWO_OR_MORE_ARGS('2', 8),
                    docstrings.RETURNS('pixel coordinates', 8))
 
+    def proj_plane_pixel_scales(self):
+        """Same as :func:`astropy.wcs.utils.proj_plane_pixel_scales`
+        but returns a list of `~astropy.units.Quantity`.
+
+        """
+        from astropy.wcs import utils as wcs_utils  # Avoid circular import
```

Comment:
Why not move the code from `utils.py` here as `_ proj_plane_pixel_scales()` and then call it from both functions? Doing this, I think, would avoid the need for this import within the method.

Reply:
@pllim I did not do any timing tests on this specific PR, but I do not think the answer from https://stackoverflow.com/a/32151422 is accurate. The benchmark does not compare timings of global import vs local import.

I did a quick test on a function that uses numpy's `sin()` to do a simple computation:

#### Local import
```python
In [1]: def f(x):
   ...:     from numpy import sin
   ...:     return sin(x)
   ...:

In [2]: %timeit f(0.5)
1.08 µs ± 10.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: exit
```

#### Global import
```
In [1]: from numpy import sin

In [2]: def f(x):
   ...:     return sin(x)
   ...:

In [3]: %timeit f(0.5)
587 ns ± 7.57 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
Let's use a more realistic example:

```python
>>> from astropy.io import fits
>>> from astropy.utils.data import get_pkg_data_filename
>>> from astropy.wcs import WCS

>>> path = get_pkg_data_filename("data/dist_lookup.fits.gz", package='astropy.wcs.tests')
>>> with fits.open(path) as hdulist:
...     w = WCS(hdulist[1].header, hdulist)

>>> # Time this first before wcs.utils import below.
>>> # This was done in IPython. I just have a habit to use >>> in markdown.
>>> %timeit w.proj_plane_pixel_scales()
87.6 µs ± 78.8 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

>>> from astropy import units as u
>>> from astropy.wcs.utils import proj_plane_pixel_scales
>>> # Same calculation as the class method
>>> def proj_this(w):
...     values = proj_plane_pixel_scales(w)
...     units = [u.Unit(x) for x in w.wcs.cunit]
...     return [value * unit for (value, unit) in zip(values, units)]

>>> # Time the standalone function
>>> # This was done in IPython. I just have a habit to use >>> in markdown.
>>> %timeit proj_this(w)
87.1 µs ± 149 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Okay, that's 0.5 microseconds improvement in something that takes 87 microseconds to calculate. Is this small improvement worth the extra code complexity in moving the `utils` functions completely out of `utils` just to avoid circular import while being able to do module level import?

Reply:
I do not think the code is more complex, it is easier to read (follow), and it is faster. I suspect that is the reason why my older phone is "getting slower" :)

Reply:
Could you, please, for completeness, benchmark `utils.proj_plane_pixel_scales(w)`. I am getting like 1/3 of your timing.

Reply:
What's the point of that? It doesn't apply units, so it is like comparing orange to apple.